gen3-client/jwt/functions.go

```diff
@@ -184,8 +180,54 @@ func (f *Functions) DoRequestWithSignedHeader(profile string, configFileType str
 		f.Config.UpdateConfigFile(cred, []byte(content), cred.APIEndpoint, configPath, profile)
 
 		resp, err = f.Request.MakeARequest(method, apiEndpoint, cred.AccessKey, contentType, bytes.NewBuffer(bodyBytes))
-		msg, err = f.ParseFenceURLResponse(resp)
-		return msg.Url, msg.GUID, err
 	}
+
+	return prefixEndPoint, resp, nil
+}
+
+func (f *Functions) DoRequestWithSignedHeader(profile string, configFileType string, endpointPostPrefix string, contentType string, bodyBytes []byte) (string, string, error) {
+	/*
+	   Do request with signed header. User may have more than one profile and use a profile to make a request
+	*/
+	var err error
+	var msg JsonMessage
+
+	_, resp, err := f.GetResponse(profile, configFileType, endpointPostPrefix, contentType, bodyBytes)
+	if err != nil {
+		return "", "", err
+	}
+
+	msg, err = f.ParseFenceURLResponse(resp)
+	return msg.Url, msg.GUID, err
+
+	panic("Unexpected case")
+}
+
+func (f *Functions) CheckPrivileges(profile string, configFileType string, endpointPostPrefix string, contentType string, bodyBytes []byte) (string, map[string]interface{}, error) {
+	/*
+	   Return user privileges from specified profile
+	*/
+	var err error
+	var data map[string]interface{}
+
+	host, resp, err := f.GetResponse(profile, configFileType, endpointPostPrefix, contentType, bodyBytes)
+	if err != nil {
+		return "", nil, err
+	}
+
+	str := ResponseToString(resp)
+
+	err = json.Unmarshal([]byte(str), &data)
+	if err != nil {
+		panic(err)
+	}
+
+	project_access, ok := data["project_access"].(map[string]interface{})
+	if !ok {
+		panic("Not possible to read user access privileges")
+	}
+
+	return host, project_access, err
+
 	panic("Unexpected case")
```

Comment:
this code also becomes unreachable

Reply:
Thanks! Fixed in the commit [33dfedf](https://github.com/uc-cdis/cdis-data-client/pull/49/commits/33dfedfc97fbef84fe5d7a2465d075d60aa6d4dd)